### ISSUE TITLE
[Notion] Can't search and create page

### ISSUE THREAD

**Issue by flywww** [raycast/extensions#4158]
### Extension

https://www.raycast.com/HenriChabrand/notion

### Description

Searching and Creating post in Notion is not working. Already set Notion token.

**Comment by christianmagill** [comment]
I get "Unexpected error occurred." on both commands. I have created an integration and added the connection to the desired database.

**Comment by csaper** [comment]
@mathieudutour I, too, am experiencing an issue seeing results via the `Search Notion` command, as well as creating pages via the `Create Database Page` command.

I have confirmed that I have created the integration correctly in Notion and added the associated Internal Integration Token to the `Notion Token` field in the extension configurations within Raycast.
<br/>
Here are screen recordings of executing each command in Raycast:

*Notion Search*
![Notion Search](https://user-images.githubusercontent.com/316563/214899609-1a544f63-3dc7-4b4e-b092-9a522dad8085.gif)

<br/>

*Create Database Page*
![Notion Create Database Page](https://user-images.githubusercontent.com/316563/214900360-e05b69c1-62e0-4e15-8387-c0fea0fcb9d8.gif)
<br/>

**Comment by christianmagill** [comment]
@mathieudutour, I've tried removing and recreating the connection and token within Notion as well as uninstalling and reinstalling the extension. My screens now mirror @csaper, no results, but I no longer get the "Unexpected error occurred." message

**Comment by christianmagill** [comment]
@mathieudutour, I've been able to set up the extensions successfully on a brand-new Notion account. I wonder if it is the size of my account that may be the issue. If there's anything I can do to help debug the issue let me know.

**Comment by HamzaM** [comment]
> @mathieudutour, I've tried removing and recreating the connection and token within Notion as well as uninstalling and reinstalling the extension. My screens now mirror @csaper, no results, but I no longer get the "Unexpected error occurred." message

Just setup raycast notion extension and I am unable to search/create pages unfortunately.

**Comment by denkasyanov** [comment]
TLDR: make sure that the required page has "Connection" set up in Notion

I had not exactly the same but similar problem.

After initial extension setup I created new database.
I tried to select this new database in "Create Database Page" but it wasn't there.

It appears, on your first login you give access to the specific set of root pages. And if the new database is under the root page that was created after the login, it will be unavailable in Raycast. You need to add a connection to make it available.

So I added it in Notion app and the new database became available immediately.

To add a connection, open required page (in my case it was the new root page) and add a connection via the menu in the top right corner.
![CleanShot 2023-04-14 at 00 21 37@2x](https://user-images.githubusercontent.com/34678129/231886680-1ce1f35c-2213-476b-b7c9-fb8ccecc4726.png)

I believe the same limitations apply not only to creating pages in databases but to searching as well.

**Comment by mathieudutour** [comment]
Thanks for the nice write up. Would you mind adding those explanations to the README of the extension?

**Comment by denkasyanov** [comment]
@mathieudutour Sorry for the late answer.

Sure!

I believe I found a little bit more robust way to address this issue.

https://github.com/raycast/extensions/pull/6375

**Comment by christianmagill** [comment]
I still think there's also a problem with larger databases. I can set things up properly in a new workspace with no problems. When I try and set things up on my larger workspace I run into issues.
